tokio/src/runtime/task/join.rs

```diff
@@ -203,6 +203,43 @@ impl<T> JoinHandle<T> {
         }
     }
 
+    /// Checks if the task associated with the handle has finished(whether completed or cancelled).
+    /// This function does not block.
+    ///
+    /// # Notes
+    ///
+    /// Cancelling a task may need some time to take effect. This function only returns true
+    /// when the task has completed the cancellation process.
```

Comment:
```suggestion
    /// Checks if the task associated with this `JoinHandle` has finished.
    ///
    /// Please note that this method can return `false` even if `abort` has been
    /// called on the task. This is because the cancellation process may take
    /// some time, and this method does not return `true` until it has
    /// completed.
```